test/integration/logins-api-test.js

```diff
@@ -0,0 +1,177 @@
+    await driver.get(webext.url("/test/integration/test-pages/logins-api.html"));
+  };
+
+  const clickButton = async (id) => {
+    await webext.inContent();
+    const btn = await driver.wait(until.elementLocated(
+      By.id(id)
+    ), 1000);
+    await btn.click();
+  };
+
+  before(async () => {
+    webext = await getWebExtension();
+    await webext.start();
+    driver = await webext.driver;
+    webdriver = webext.webdriver;
+    By = webdriver.By;
+    until = webdriver.until;
+    await webext.inChrome();
+    await driver.executeScript(`
+      ChromeUtils.defineModuleGetter(this, "LoginHelper",
+                                     "resource://gre/modules/LoginHelper.jsm");
+      ChromeUtils.defineModuleGetter(this, "Services",
+                                     "resource://gre/modules/Services.jsm");
+    `);
+  });
+
+  beforeEach(async () => {
+    await webext.inChrome();
+    await driver.executeScript(`
+      Services.logins.removeAllLogins();
+    `);
+  });
+
+  after(async () => {
+    await driver.quit();
+  });
+
+  it("browser.experiments.logins.add should add a new login", async () => {
+    await webext.inChrome();
+    const initialLogins = await getLogins();
+    expect(initialLogins).to.be.an("array").that.is.empty;
+
+    await loadTestPage();
+    await clickButton("add");
+
+    await webext.inChrome();
+    const finalLogins = await getLogins();
+    expect(finalLogins[0]).to.deep.equal(mockLogin);
+  });
+
+  it("browser.experiments.logins.remove should remove the login", async () => {
+    await addLogin();
+
+    await loadTestPage();
+    await clickButton("remove");
+
+    await webext.inChrome();
+    const results = await getLogins();
+    expect(results).to.be.an("array").that.is.empty;
+  });
+
+  it("browser.experiments.logins.get should return the login", async () => {
+    await addLogin();
+
+    await loadTestPage();
+    await clickButton("get");
+
+    const results = await driver.wait(until.elementLocated(
+      By.id("get-results")
+    ), 1000);
+    await driver.wait(until.elementTextContains(results, "guid"), 5000);
+    const output = await results.getText();
+    expect(JSON.parse(output)).to.deep.equal(mockLogin);
+  });
+
+  it("browser.experiments.logins.getAll should return the login in an array", async () => {
+    await addLogin();
+
+    await loadTestPage();
+    await clickButton("get-all");
+
+    const results = await driver.wait(until.elementLocated(
+      By.id("get-all-results")
+    ), 1000);
+    await driver.wait(until.elementTextContains(results, "guid"), 5000);
+    const output = await results.getText();
+    expect(JSON.parse(output)).to.deep.equal([mockLogin]);
+  });
+
+  it("browser.experiments.logins.update should update the username", async () => {
+    await addLogin();
+
+    await loadTestPage();
+    await clickButton("update");
+
+    const logins = await getLogins();
+    expect(logins[0].username).to.equal(mockLogin.username);
```

Comment:
Oh, this is wrong. The username shouldn't match the initial value. I'll fix.

Reply:
I've got something wrong with the `update()` method. Clearing review requests for now.